_posts/2018-02-15-upgrading-a-monolith.markdown

```diff
@@ -0,0 +1,19 @@
+---
+layout: post
+title: "Upgrading a monolith from Rails 4.0 to 5.1"
+date: 2018-02-15
+categories: ["rails", "upgrades", "case-study"]
+author: "emily"
+---
+
+We recently completed work with [Power Home Remodeling](https://powerhrg.com) on a Rails upgrade for their self-described “monolith CRM/ERP application”, and were able to speak to them about their experience with Ombu Labs.
+
+<!--more-->
+
+We talked to [Ben Langfeld](https://www.linkedin.com/in/benlangfeld/), the Application Solution Architect at [Power Home Remodeling](https://powerhrg.com), about the work on their app, Nitro. According to Ben, their monolithic CRM/ERP application is continuously built by a team of approximately 50 developers, system administrators, testers and support staff. The application contains *over 500,000 lines of Ruby on Rails and Javascript code*, and by their own definition, is one of the more complex Rails applications out there in terms of scope.
+
```

Comment:
@ercohen14 Maybe in the middle we can add a paragraph explaining a little bit about the architecture of Nitro? @mauro-oto can help with this and add more info about their http://shageman.github.io/cbra.info/ setup.

Reply:
@etagwerker @ercohen14 maybe something like:

```
... is one of the more complex Rails applications out there in terms of scope.

Nitro is a [Component-Based Rails Application](http://shageman.github.io/cbra.info/). CBRA is an approach which organizes Rails applications into Ruby Gems and Rails Engines. By using this approach, they're able to parallelize their test suite, therefore making CI faster, and reduce merge conflicts due to the large size of their team. Also, they can onboard new developers faster by having the ability to separate into teams based on gems/engines instead of having to work their way through the entire codebase.  

We executed a full upgrade of the application from Rails 4.0 to 5.1...
```

Also, you could maybe add a link somewhere to a talk by Ben Klang on CBRA: https://www.youtube.com/watch?v=tkL9On9HVHQ